src/components/toolbar/helpers.ts

```diff
@@ -207,41 +212,70 @@ export const promoteParagraphToSection = (
     offset += precedingSection.nodeSize
   }
 
-  const textContent = paragraph.textContent
+  const textContent = findChildrenByType(paragraph, schema.nodes.text).map(
+    ({ node }) => node
+  )
 
   const sectionTitle: SectionTitleNode = textContent
-    ? nodes.section_title.create({}, schema.text(textContent))
+    ? nodes.section_title.create({}, Fragment.from(textContent))
     : nodes.section_title.create()
 
   let sectionContent = Fragment.from(sectionTitle)
   if (endIndex < parentSection.childCount) {
     sectionContent = sectionContent.append(
       parentSection.content.cut(afterParagraphOffset)
     )
-  } else {
-    sectionContent = sectionContent.append(Fragment.from(paragraph.copy()))
   }
 
   if (parentSection.type.name === 'body') {
-    const newSection = nodes.section.create({}, sectionContent)
-    items[0] = nodes.body.create(
-      {},
-      items[0]
-        ? items[0].content.append(Fragment.from(newSection))
-        : Fragment.from(newSection)
-    )
+    const resolvePos = findChildrenByType(parentSection, nodes.section)[0]
+    if (resolvePos) {
+      const { pos } = resolvePos
+      items[0] = nodes.section.create(
+        {},
+        Fragment.from(sectionTitle).append(
+          parentSection.slice(afterParagraphOffset, pos).content
```

Comment:
Could you store this in a sectionContent variable then after if else statement do the:

items[0] = nodes.section.create({}, Fragment.from(sectionTitle).append(sectionContent))

to avoid duplication of the nodes.section.create(...) call.